File: C++/src/longest_common_prefix.cpp
#include <longest_common_prefix.hpp>
using namespace std;

string Solution::longestCommonPrefix(vector<string> &strs) {
  string longestPrefix = strs.empty() ? "" : strs[0];

  for (size_t i = 1; i < strs.size(); i++) {
    while (strs[i].find(longestPrefix) != 0) {
      longestPrefix.pop_back();
      if (longestPrefix.empty()) {
        return "";
      }
    }
  }

  return longestPrefix;
}

File: JavaScript/src/longest_common_prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
function longestCommonPrefix(strs) {
  let longestPrefix = strs[0] || "";

  for (let i = 0; i < strs.length; i++) {
    while (strs[i].indexOf(longestPrefix) !== 0) {
      longestPrefix = longestPrefix.slice(0, longestPrefix.length - 1);
      if (longestPrefix === "") {
        return "";
      }
    }
  }

  return longestPrefix;
}

export { longestCommonPrefix };
